### DISCUSSION TITLE
Accept plugin configuration in pytest.ini

### DISCUSSION THREAD

**Discussion by pawamoy** [pytest-dev/pytest#13078] [Q&A] [answered]
Is it possible for a plugin writer to get its configuration from pytest.ini?

For example:

```ini
[pytest]
python_files =
  test_*.py
testpaths =
  tests

[inline-snapshot]
snapshot_dir = tests/snapshots
```

Then in `pytest_configure`:

```python
def pytest_configure(config):
    my_config = config.getini("inline-snapshot")
```

```
ValueError("unknown configuration value: 'inline-snapshot'")
```

Currently it seems like the only way is to declare CLI options, and then for the user to add these options to `addopts`, which has the disadvantage of only supporting strings (and therefore requires that these strings be coerced into something else by the plugin).

But I'm a noob regarding pytest plugins and their config so I might be missing a few things :slightly_smiling_face:

**Comment by pawamoy** [comment]
I suppose plugins could re-read the config file to load their own settings:

```python
def pytest_configure(config):
    iniconfig = configparser.ConfigParser()
    iniconfig.read(config.inifile)
    my_config = iniconfig["inline-snapshot"]
```

Would you recommend that, or is there a better way?

**Comment by The-Compiler** [comment, marked as answer]
You will need to call [`Parser.addini()`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.Parser.addini) in [`pytest_addoption`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_addoption) to declare your options, before you use [`Config.getini()`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.Config.getini).

That doesn't let you define an entirely new section, however. Reading the ini yourself is definitely something I wouldn't recommend, as you probably won't be compatible with pytest configuration in `pyproject.toml`, and might have possible subtle syntax differences to pytest's ini parsing.

**Reply by pawamoy** [reply]
Thanks @The-Compiler! That's what I was missing :slightly_smiling_face:
